core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/PhysicalServices.java

```diff
@@ -847,11 +849,39 @@ public void updatePhysicalPathStyles(DDiagram diagram) {
         }
       }
     }
+    
+    customizePhyscialLinkEdgeLabels(coloredLinks, displayedPaths);
 
     // Reset physical links with no physical path
     for (DEdge aPL : displayedPhysicalLinks.values()) {
       if (!coloredLinks.containsKey(aPL)) {
         resetPhysicalLinkStyle(aPL);
+        resetPhysicalLinkEdgeLabels(aPL);
+      }
+    }
+  }
+
+  public void resetPhysicalLinkEdgeLabels(DEdge aPL) {
+    DEdgeIconCache.getInstance().removeIcon(aPL);
+    DiagramServices.getDiagramServices().refreshBeginEndLabels(aPL);
+  }
+
+  /**
+   * Add pie icons on the begin and the end labels of a Physical Link
+   * 
+   * @param coloredLinks
+   * @param displayedPaths
+   */
+  public void customizePhyscialLinkEdgeLabels(Map<DEdge, Set<PhysicalPath>> coloredLinks,
+      Map<PhysicalPath, DNode> displayedPaths) {
+    for (Map.Entry<DEdge, Set<PhysicalPath>> entry : coloredLinks.entrySet()) {
+      DEdge edge = entry.getKey();
+      Set<PhysicalPath> paths = entry.getValue();
+      Set<RGBValues> pathColors = paths.stream().map(displayedPaths::get).map(ShapeUtil::getNodeColorStyle)
```

Comment:
not sure about set,
if we close session and reopen, maybe order of paths are differents, computing another pie ?
wdyt?

Reply:
Effectivelty, I was thinking about not changing the content of icons in diagrams with the same paths. Otherwise we can see differences in diagram when comparing a model and its copy in the same workspace for example.

Reply:
Finally I changed Set to List and reorder the list of colors. That way we'll always have the same icon even we restart the app.